arangod/Aql/IResearchViewNode.cpp

```diff
@@ -681,28 +681,34 @@ ViewSnapshotPtr snapshotSingleServer(IResearchViewNode const& node,
     if (options.forceSync) {
       syncViewSnapshot(*snapshot, view.name());
     }
-    if (options.restrictSources) {
-      return std::make_shared<ViewSnapshotView>(*snapshot, options.sources);
-    }
+    // if (options.restrictSources) {
+    //  return std::make_shared<ViewSnapshotView>(*snapshot, options.sources);
+    //}
     return {ViewSnapshotPtr{}, snapshot};
   }
-  if (!options.forceSync && trx.isMainTransaction()) {
+
+  // TODO: remove here restrincSources condition when switching back to
+  // ViewSnapshotView
+  if (!options.restrictSources && !options.forceSync &&
+      trx.isMainTransaction()) {
     return {};
   }
   auto links = [&] {
     if (view.type() == ViewType::kArangoSearch) {
       auto const& viewImpl = basics::downCast<IResearchView>(view);
-      return viewImpl.getLinks();
+      return viewImpl.getLinks(options.restrictSources ? &options.sources
+                                                       : nullptr);
     } else {
       auto const& viewImpl = basics::downCast<Search>(view);
-      return viewImpl.getLinks();
+      return viewImpl.getLinks(options.restrictSources ? &options.sources
+                                                       : nullptr);
     }
   }();
   snapshot = makeViewSnapshot(trx, key, options.forceSync, view.name(),
                               std::move(links));
-  if (options.restrictSources && snapshot != nullptr) {
-    return std::make_shared<ViewSnapshotView>(*snapshot, options.sources);
-  }
+  // if (options.restrictSources && snapshot != nullptr) {
+  //  return std::make_shared<ViewSnapshotView>(*snapshot, options.sources);
+  //}
```

Comment:
Still needed?

Reply:
We want to restore Snapshot "views" in future PRs. Just now it is broken :( , so we disabled this code temporary